recipes/wineditline/all/conanfile.py

```diff
@@ -1,64 +1,62 @@
-import functools
 import os
 
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.53.0"
 
 
 class WineditlineConan(ConanFile):
     name = "wineditline"
-    description = (
-        "A BSD-licensed EditLine API implementation for the native "
-        "Windows Console"
-    )
+    description = "A BSD-licensed EditLine API implementation for the native Windows Console"
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "http://mingweditline.sourceforge.net/"
     topics = ("readline", "editline", "windows")
-    license = "BSD-3-Clause"
-    generators = ("cmake",)
-    settings = ("os", "arch", "compiler", "build_type")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
     }
     default_options = {
         "shared": False,
     }
-    exports_sources = ("patches/*", "CMakeLists.txt")
+    provides = "editline"
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", dst=self.export_sources_folder, src=self.recipe_folder)
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
```

Comment:
src_folder is not needed here, works fine without it. Root folder contains CMakeLists

Reply:
This is a CCI convention, src is expected to be the source folder for every package :)